### PR TITLE
feat: add update sdk action

### PR DESCRIPTION
**Pull request by helciofranco**
Add a reusable workflow to automatically update SDK packages for a given list of packages.

| Action  | Result |
| ----- | ------- |
| Screenshot 2024-03-18 at 19 36 08 | Screenshot 2024-03-18 at 20 06 08 |

### How to use?

```yaml
- uses: FuelLabs/github-actions/update-sdk@master
  with:
    branch: master # In case you want to use a different head branch, default is "main"
    packages: |
      @fuels
      @fuels/react
      @fuels/connectors
    npm-tag: latest # You might use "latest" or anything else
  env:
    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

### CODE REVIEW

File: update-sdk/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    npm-tag: ${{ matrix.tag }} # You might use "latest" or anything else, like a matrix strategy
+  env:
+    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+```
+
+### Inputs
+
+| Name         | Description                                                |
+| ------------ | ---------------------------------------------------------- |
+| npm-tag      | NPM tag (e.g. latest or next)                              |
```
Comment (luizstacio) [inline update-sdk/README.md:24]: We should add the other inputs for repository, branch....

File: update-sdk/action.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+name: "SDK Outdated"
+description: "Reusable workflow for update the SDK packages"
+
+inputs:
+  repository:
+    description: "Github Repository"
+    default: ${{ github.repository }}
+  branch:
+    description: "Branch name"
```
Comment (luizstacio) [inline update-sdk/action.yaml:9]: We should be more descriptive here, like `The name of the branch to be created`

File: update-sdk/src/Github.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    console.log(c.white(`📤 Pushing branch ${head}`));
+    await $`gh repo set-default ${this.repository}`;
+    const { stdout } = await $`gh pr create --base ${base} --head ${head} --title ${title} --body ${body}`;
+    console.log(c.green(`✅ PR created: ${stdout}`));
+  }
+
+  async setupGitAgent() {
+    await $`git config --global user.email "github-actions[bot]@users.noreply.github.com"`;
+    await $`git config --global user.name "github-actions[bot]"`;
+    await $`git config --global push.autoSetupRemote true`;
```
Comment (luizstacio) [inline update-sdk/src/Github.ts:57]: Do you know if this will work with a PAT token?

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: update-sdk/action.yaml (hunk 2 above)
Comment (helciofranco) [reply update-sdk/action.yaml]: https://github.com/FuelLabs/github-actions/pull/26/commits/30378159fa1e43b65545271ec435ecd01570def9

File: update-sdk/README.md (hunk 1 above)
Comment (helciofranco) [reply update-sdk/README.md]: https://github.com/FuelLabs/github-actions/pull/26/commits/30378159fa1e43b65545271ec435ecd01570def9

File: update-sdk/src/Github.ts (hunk 3 above)
Comment (helciofranco) [reply update-sdk/src/Github.ts:61]: Yes, I've tested with my PAT (you just need to have `write:repo` permissions)

**Review by luizstacio — APPROVED** [review]

**Pull request merged**